subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -1692,34 +1748,259 @@ int bt_cap_initiator_unicast_audio_stop(const struct bt_cap_unicast_audio_stop_p
 
 	for (size_t i = 0U; i < param->count; i++) {
 		struct bt_cap_stream *cap_stream = param->streams[i];
+		struct bt_bap_stream *bap_stream = &cap_stream->bap_stream;
+
+		/* Ensure that ops are registered before any procedures are started */
+		bt_cap_stream_ops_register_bap(cap_stream);
 
 		active_proc->proc_param.initiator[i].stream = cap_stream;
-	}
+		active_proc->proc_param.initiator[i].stop.release = param->release;
 
-	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_STOP, param->count);
+		if (can_disable_stream(bap_stream)) {
+			can_disable = true;
+		}
 
-	bt_cap_common_set_subproc(BT_CAP_COMMON_SUBPROC_TYPE_RELEASE);
+		if (can_stop_stream(bap_stream)) {
+			can_disable = true;
+		}
 
+		if (param->release && can_release_stream(bap_stream)) {
+			can_release = true;
+		}
+	}
+
+	if (!can_disable && !can_stop && !can_release) {
```

Comment:
Does -EALREADY make sense here?

The GNU spec. states it as:
'EALREADY
The socket socket is non-blocking and already has a pending connection in progress (see EINPROGRESS above).'

And while its true that we "already has a pending connection in progress", I think the intent of the wording is when we try to connect a new one.

Could EBUSY make more sense here?

Reply:
The case is here is that all streams are already stopped, and thus `-EBUSY` doesn't make any sense. 

The alternative is that we return `0`, but then it's hard for an application to determine whether a request was successfully sent, or if nothing was done. 

We need a way to tell the caller that no request was sent because it would not result in any state changes

Reply:
I guess this make sense. In regardless, I certainly think '0' would be wrong here :D